types/airbnb-prop-types/index.d.ts

```diff
@@ -24,6 +24,12 @@ export type ReactFunctionComponentLike = (...args: any[]) => PropTypes.ReactNode
 
 export type ReactTypeLike = string | ReactClassComponentLike | ReactFunctionComponentLike;
 
+export interface ReactRefLike<T> {
+    readonly current: T | null;
+}
+
+export type ReactLegacyRefLike<T> = (instance: T | null) => void | ReactRefLike<T>;
```

Comment:
I think you want `((instance: T | null) => void) | ReactRefLike<T>`, which is a union of either. Currently it's a return type of either.

Reply:
Yeah, you're right. Nice catch.

I avoided this while testing because the order was flipped